modules/queue/src/main/scala/ch/epfl/bluebrain/nexus/service/queue/KafkaPublisher.scala

```diff
@@ -0,0 +1,83 @@
+package ch.epfl.bluebrain.nexus.service.queue
+
+import java.util.concurrent.Future
+
+import akka.NotUsed
+import akka.actor.{ActorRef, ActorSystem}
+import akka.kafka.ProducerMessage._
+import akka.kafka.ProducerSettings
+import akka.kafka.scaladsl.Producer
+import akka.persistence.query.Offset
+import akka.stream.scaladsl.Flow
+import cats.Show
+import cats.syntax.show._
+import ch.epfl.bluebrain.nexus.service.indexer.persistence.SequentialTagIndexer
+import io.circe.Encoder
+import io.circe.syntax._
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord, RecordMetadata}
+import shapeless.Typeable
+
+/**
+  * Helper class to send messages to different Kafka topics sharing only one instance
+  * of a [[org.apache.kafka.clients.producer.KafkaProducer]]
+  *
+  * @param producer the shared [[KafkaProducer]] instance
+  * @param topic    the kafka topic
+  * @tparam Event   the generic event type
+  */
+class KafkaPublisher[Event: Encoder: Show](producer: KafkaProducer[String, String], topic: String) {
+
+  /**
+    * Manually sends a single message
+    *
+    * @param event the event to send
+    * @return the metadata for the record acknowledge by the server
+    */
+  def send(event: Event): Future[RecordMetadata] = {
+    val message = new ProducerRecord[String, String](topic, event.show, event.asJson.noSpaces)
+    producer.send(message)
+  }
+}
+
+object KafkaPublisher {
+
+  private def flow[Event: Encoder: Show](producerSettings: ProducerSettings[String, String],
+                                         topic: String): Flow[(Offset, String, Event), Offset, NotUsed] = {
+    Flow[(Offset, String, Event)]
+      .map {
+        case (off, _, event) =>
+          Message(new ProducerRecord[String, String](topic, event.show, event.asJson.noSpaces), off)
+      }
+      .via(Producer.flexiFlow(producerSettings))
+      .map(_.passThrough)
+
+  }
+  // $COVERAGE-OFF$
+  /**
+    * Starts publishing events to Kafka using a [[ch.epfl.bluebrain.nexus.service.indexer.persistence.SequentialTagIndexer]]
+    *
+    * @param projectionId projection to user for publishing events
+    * @param pluginId query plugin ID
+    * @param tag events with which tag to publish
+    * @param name name of the actor
+    * @param producerSettings Akka StreamsKafka producer settings
+    * @param topic topic to publish to
+    * @param as implicit ActorSystem
+    * @tparam Event the generic event type
+    * @return ActorRef for the started actor
+    */
+  final def start[Event: Encoder: Show: Typeable](projectionId: String,
```

Comment:
should we rename this to: `startTagStream` to make it easier to understand what it does?